pkg/kube/translation/gateway_httproute.go

```diff
@@ -52,25 +52,46 @@ func (t *translator) TranslateGatewayHTTPRouteV1Alpha2(httpRoute *gatewayv1alpha
 		for j, backend := range backends {
 			//TODO: Support filters
 			//filters := backend.Filters
-			kind := strings.ToLower(string(*backend.Kind))
+			var kind string
+			if backend.Kind == nil {
+				kind = "service"
+			} else {
+				kind = strings.ToLower(string(*backend.Kind))
+			}
 			if kind != "service" {
 				log.Warnw(fmt.Sprintf("ignore non-service kind at Rules[%v].BackendRefs[%v]", i, j),
 					zap.String("kind", kind),
 				)
 				continue
 			}
 
-			ns := string(*backend.Namespace)
+			var ns string
+			if backend.Namespace == nil {
+				ns = httpRoute.Namespace
+			} else {
+				ns = string(*backend.Namespace)
+			}
 			//if ns != httpRoute.Namespace {
 			// TODO: check gatewayv1alpha2.ReferencePolicy
 			//}
 
+			if backend.Port == nil {
+				log.Warnw(fmt.Sprintf("ignore nil port at Rules[%v].BackendRefs[%v]", i, j),
+					zap.String("kind", kind),
+				)
+				continue
+			}
+
 			ups, err := t.TranslateUpstream(ns, string(backend.Name), "", int32(*backend.Port))
 			if err != nil {
 				return nil, errors.Wrap(err, fmt.Sprintf("failed to translate Rules[%v].BackendRefs[%v]", i, j))
 			}
 			name := apisixv1.ComposeUpstreamName(ns, string(backend.Name), "", int32(*backend.Port))
-			ups.Labels["id-name"] = name
+
+			ups.Labels["meta_namespace"] = truncate(ns, 64)
```

Comment:
We can add a comment explaining why 64 is used  https://github.com/apache/apisix/blob/5b95b85faea3094d5e466ee2d39a52f1f805abbb/apisix/schema_def.lua#L85